backend/de.metas.adempiere.adempiere/base/src/main/java/de/metas/report/FallbackDocumentReportAdvisor.java

```diff
@@ -102,10 +107,19 @@ public DocumentReportInfo getDocumentReportInfo(
 			reportProcessId = util.getReportProcessIdByPrintFormatId(printFormatId);
 		}
 
+		final Integer locationId = InterfaceWrapperHelper.getValueOrNull(record, COLUMNNAME_C_BPartner_Location_ID);
+		final BPartnerLocationId bPartnerLocationId = BPartnerLocationId.ofRepoIdOrNull(bpartnerId, locationId);
```

Comment:
might get NPE when locationId is null

Reply:
also i would suggest introducing DocumentReportAdvisorUtil.getBPartnerLocationId method, similar to getBPartnerId and others...

Reply:
Why? ofRepoIdOrNull() checks it

Reply:
> also i would suggest introducing DocumentReportAdvisorUtil.getBPartnerLocationId method, similar to getBPartnerId and others...

done